src/dbnode/persist/fs/msgpack/decoder.go

```diff
@@ -111,12 +117,14 @@ func (dec *Decoder) DecodeIndexInfo() (schema.IndexInfo, error) {
 }
 
 // DecodeIndexEntry decodes index entry
-func (dec *Decoder) DecodeIndexEntry(bytesPool pool.BytesPool) (schema.IndexEntry, error) {
+func (dec *Decoder) DecodeIndexEntry(bytesPool pool.BytesPool, validate bool) (schema.IndexEntry, error) {
```

Comment:
Do we actually need this validate bool? Perhaps instead we should unconditionally validate if the field is present at all times?

Makes the API more simple I believe?

Reply:
Yeah, I think you're right. Probably OK if there's a small subset of files that do full file checksumming + entry level checksumming for the sake of a cleaner API.